client_server_test.go

```diff
@@ -659,3 +661,31 @@ func TestSocksProxyDial(t *testing.T) {
 	defer ws.Close()
 	sendRecv(t, ws)
 }
+
+func TestDialWithContext(t *testing.T) {
+	s := newServer(t)
+	defer s.Close()
+
+	var headersWrote, requestWrote bool
+	trace := &httptrace.ClientTrace{
```

Comment:
Test for all relevant events and implement missing events as needed.  Events that seem relevant are GetConn, GotConn, ConnectDone, TLSHandshakeStart, TLSHandshakeDone,  WroteHeaders, WroteRequest, GotFirstResponseByte

Reply:
TLSHandshakeStart and TLSHandshakeDdone are available only since 1.8. Do you want me to handle it with conditionnal build?